src/actions.ts

```diff
@@ -1,21 +1,28 @@
-import { fetchOPDSData } from './fetchData';
+import { fetchOPDSData } from "./fetchData";
 
-export const REQUEST_COLLECTION = 'REQUEST_COLLECTION';
-export const LOAD_COLLECTION = 'LOAD_COLLECTION';
+export const FETCH_COLLECTION_REQUEST = "FETCH_COLLECTION_REQUEST";
+export const FETCH_COLLECTION_SUCCESS = "FETCH_COLLECTION_SUCCESS";
+export const LOAD_COLLECTION = "LOAD_COLLECTION";
 
 export function fetchCollection(url: string) {
   return function(dispatch) {
-    dispatch(requestCollection(url));
-    console.log("fetching url: ", url)
-
-    return fetchOPDSData(url, (data) => {
-      dispatch(loadCollection(data, url));
+    dispatch(fetchCollectionRequest(url));
+    return new Promise((resolve, reject) => {
+      fetchOPDSData(url).then((data: CollectionProps) => {
+        dispatch(fetchCollectionSuccess());
+        dispatch(loadCollection(data, url));
```

Comment:
I don't remember if you said why you were splitting up these actions, but now I'm not sure it makes sense. We wouldn't want the loading message to go away before the data is there. Maybe they should be separate but we should check a different property to decide whether to show the loading indicator.